signer/migrations/0012__dkg_verification_extensions.sql

```diff
@@ -1,75 +1,52 @@
--- Enum table for DKG shares status
-CREATE TABLE sbtc_signer.dkg_shares_status (
-    -- The id of the status, not auto-incremented as we want to control the values.
-    id INTEGER PRIMARY KEY,
-    -- The name of the status.
-    key TEXT NOT NULL,
-    -- Brief description of what the status means.
-    description TEXT NOT NULL
-);
 
--- Insert the initial entries.
-INSERT INTO sbtc_signer.dkg_shares_status (id, key, description) VALUES
-    (0, 'PENDING', 'DKG round successful, pending verification via signing round'),
-    (1, 'VERIFIED', 'Successfully verified via signing round'),
-    (2, 'KEY_REVOKED', 'The DKG key has been revoked and should not be used');
+CREATE TYPE sbtc_signer.dkg_shares_status AS ENUM (
+    'unverified',
+    'verified',
+    'failed'
+);
 
 -- Add the new columns to the `dkg_shares` table. We're not adding indexes for
 -- now because the table is so small that the overhead likely outweighs the
 -- benefits.
 ALTER TABLE sbtc_signer.dkg_shares
-    -- Contains the current 
-    ADD COLUMN dkg_shares_status_id INTEGER DEFAULT 0 REFERENCES sbtc_signer.dkg_shares_status (id),
-    ADD COLUMN verified_at_bitcoin_block_hash BYTEA DEFAULT NULL,
-    ADD COLUMN verified_at_bitcoin_block_height BIGINT DEFAULT NULL,
-    ADD CONSTRAINT fk_dkg_shares_bitcoin_block_hash
-        FOREIGN KEY (verified_at_bitcoin_block_hash)
-        REFERENCES sbtc_signer.bitcoin_blocks (block_hash),
-    ADD CONSTRAINT chk_verified_at
-        CHECK (
-            (dkg_shares_status_id = 1 AND verified_at_bitcoin_block_hash IS NOT NULL AND verified_at_bitcoin_block_height IS NOT NULL) 
```

Comment:
These check constraints (if modified slightly after the enum change) ensure that the status can't be `verified` without these also being present, and vice versa. I think we should keep them from a data integrity perspective.

Reply:
Ah, nevermind, you added not null below so this can never happen since we write at start block 👍